Support Running Unit Tests With `wp-env` (#37799)
woocommerce/woocommerce: plugins/woocommerce/tests/README.md

## Table of contents

- [WooCommerce Tests](#woocommerce-tests)
    - [Table of contents](#table-of-contents)
    - [Recommended Environment](#recommended-setup)
    - [Manual Environment](#manual-setup)
        - [MySQL database](#mysql-database)
        - [Setup instructions](#setup-instructions)
    - [Running Tests](#running-tests)
        - [Troubleshooting](#troubleshooting)
    - [Writing Tests](#writing-tests)
    - [Automated Tests](#automated-tests)
    - [Code Coverage](#code-coverage)

## Recommended Setup

> We provide support for running unit tests within a `wp-env` container. All you have to do is complete the [Getting Started Guide](https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce#getting-started) and have a functional `wp-env` environment.

Reviewer: >All you have to do is

Is this a bit reductive? 

>and have a functional `wp-env` environment

I don't thing there are steps for this, like `npm -g i @wordpress/env`, so maybe reword a bit.

```suggestion
Complete the [Getting Started Guide](https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce#getting-started) and you will have a functional `wp-env` environment to run unit tests.
```

## Running Tests

Depending your environment, you will use one of these commands to run unit tests:

- **Recommended Environment**: `pnpm --filter=woocommerce test:unit:env`
- **Manual Environment**: `pnpm --filter=woocommerce test:unit`

The tests will execute and you will be presented with a summary.

> You can run specific test files by providing the path to the test class, such as `pnpm --filter=woocommerce test:unit:env tests/legacy/unit-tests/importer/product.php`. A text code coverage summary can be displayed using the `--coverage-text` option.

Reviewer: Is this the same as the testing instructions?

>pnpm --filter=woocommerce test:unit:env --filter=test_handle

There is no `--filter` bit.

Running `pnpm --filter=woocommerce test:unit:env tests/legacy/unit-tests/importer/product.php` produces an error for me:

```
Cannot open file "tests/legacy/unit-tests/importer/product.php".
```

Author: For the `--filter` it should run 10 tests instead of the thousands. You're passing `--filter=test_handle` to PHPUnit which then, well, filters.

As for the file path, you're right, that doesn't seem to work. I think it's because the working directory might not be great, let me look into that.

Author: It looks like the path to a specific file isn't going to work right now. We're going to need an upstream PR to set the working directory when using `wp-env run` 😓 If we brand this as experimental I think we can still keep it. It isn't used anywhere except manually.
